pkg/workloads/deployment.go

```diff
@@ -32,73 +31,70 @@ func newDeployment(objMeta metav1.ObjectMeta, spec appsv1.DeploymentSpec) *appsv
 	}
 }
 
+func RequestDeployment(request DeploymentRequest) (*appsv1.Deployment, error) {
+	var (
+		mutationErr error
+	)
+	deployment := newDeployment(request.ObjectMeta, request.Spec)
+
+	if len(request.Mutations) > 0 {
+		for _, mutation := range request.Mutations {
+			err := mutation(nil, deployment, request.Client)
+			if err != nil {
+				mutationErr = err
+			}
+		}
+		if mutationErr != nil {
+			return deployment, fmt.Errorf("RequestDeployment: one or more mutation functions could not be applied: %s", mutationErr)
+		}
+	}
+
+	return deployment, nil
+}
+
+// CreateDeployment creates the specified Deployment using the provided client.
 func CreateDeployment(deployment *appsv1.Deployment, client cntrlClient.Client) error {
-	return client.Create(context.TODO(), deployment)
+	return resource.CreateObject(deployment, client)
 }
 
 // UpdateDeployment updates the specified Deployment using the provided client.
 func UpdateDeployment(deployment *appsv1.Deployment, client cntrlClient.Client) error {
-	_, err := GetDeployment(deployment.Name, deployment.Namespace, client)
-	if err != nil {
-		return err
-	}
-
-	if err = client.Update(context.TODO(), deployment); err != nil {
-		return err
-	}
-	return nil
+	return resource.UpdateObject(deployment, client)
 }
 
+// DeleteDeployment deletes the Deployment with the given name and namespace using the provided client.
+// It ignores the "not found" error if the Deployment does not exist.
```

Comment:
I find it a bit confusing. Can we rephrase this? Maybe something like 
```suggestion
// It returns the "not found" error if the Deployment does not exist.
```

Reply:
Thanks, I removed all the comments that said "not found" errors are ignored